src/servers/trigger.ts

```diff
@@ -265,14 +267,36 @@ const listenerHodlInvoice = ({
   })
 }
 
-const listenerExistingHodlInvoices = async (lnd: AuthenticatedLnd) => {
+const listenerExistingHodlInvoices = async ({
+  lnd,
+  pubkey,
+}: {
+  lnd: AuthenticatedLnd
+  pubkey: Pubkey
+}) => {
   const lndService = LndService()
   if (lndService instanceof Error) return lndService
 
   const invoices = await lndService.listInvoices(lnd)
   if (invoices instanceof Error) return invoices
 
   for (const lnInvoice of invoices) {
+    if (lnInvoice.isHeld) {
+      const invoicesRepo = WalletInvoicesRepository()
+      const walletInvoice = await invoicesRepo.findByPaymentHash(lnInvoice.paymentHash)
+      if (
+        walletInvoice instanceof Error ||
```

Comment:
I guess there could be some transient error here (like mongodb is offline) but that's probably ok. we would anyway decline the invoice anyhow in this scenario if mongodb were to be offline because we would not process the invoice immediately and cancel the invoice on a later cronjob

Reply:
This one is more to allow us to not decline btc-only invoices (`walletInvoice.recipientWalletDescriptor.currency !== WalletCurrency.Btc`) if we receive the event here but the invoice has been held for some time.

It would decline non-BTC invoice, but can still settle btc-only invoices if for example the trigger were to restart and it finds held invoices in lnd.